sdk/servicebus/service-bus/src/constructorHelpers.ts

```diff
@@ -48,7 +48,12 @@ export function createConnectionContextForConnectionString(
   config.webSocketEndpointPath = "$servicebus/websocket";
   config.webSocketConstructorOptions = options?.webSocketOptions?.webSocketConstructorOptions;
 
-  const credential = new SharedKeyCredential(config.sharedAccessKeyName, config.sharedAccessKey);
+  const credential = SharedKeyCredential.fromConnectionString(connectionString);
+
+  // TODO: silly workaround for purely SAS credentials.
+  config.sharedAccessKey = config.sharedAccessKey ?? credential.key;
```

Comment:
@ramya-rao-a , @HarshaNalluru  - not particularly happy about that. @ramya-rao-a  floated the idea of doing a core-amqp v2 and we could change this there but it doesn't seem worth it.

Reply:
Seems fine.
Though for v2, would we change `credential.key` to `credential.sharedAccesKey`?

Reply:
I think I'd just remove the field being exported from connection config (if possible). I think this is just one of those cases where we're overzealous in what we expose and have exposed an implementation detail.

Reply:
We've left the overall structures the same but I altered the validation function allow SAS tokens to leave the sharedKey fields blank.